### PR TITLE
Fix dataMimeType and metadataMimeType mixup in precondition check

### PR DESCRIPTION
**Pull request by yschimke**
Mainly raising the issue I think I hit in debugging, so don't mind if someone takes this over.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@yschimke Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-framework&pullRequestId=23292)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-framework&pullRequestId=23292) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@yschimke Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-framework&pullRequestId=23292)!

File: spring-messaging/src/main/java/org/springframework/messaging/rsocket/annotation/support/RSocketMessageHandler.java [hunk 1]
```diff
@@ -306,7 +306,7 @@ private MessagingRSocket createResponder(ConnectionSetupPayload setupPayload, RS
 
 		s = setupPayload.metadataMimeType();
 		MimeType metaMimeType = StringUtils.hasText(s) ? MimeTypeUtils.parseMimeType(s) : this.defaultMetadataMimeType;
-		Assert.notNull(dataMimeType, "No `metadataMimeType` in ConnectionSetupPayload and no default value");
+		Assert.notNull(metadataMimeType, "No `metadataMimeType` in ConnectionSetupPayload and no default value");
```
Comment (sbrannen) [inline spring-messaging/src/main/java/org/springframework/messaging/rsocket/annotation/support/RSocketMessageHandler.java]:
```suggestion
		Assert.notNull(metaMimeType, "No `metadataMimeType` in ConnectionSetupPayload and no default value");
```

**Review by sbrannen — CHANGES_REQUESTED** [review]

File: spring-messaging/src/main/java/org/springframework/messaging/rsocket/annotation/support/RSocketMessageHandler.java (hunk 1 above)
Comment (sbrannen) [reply spring-messaging/src/main/java/org/springframework/messaging/rsocket/annotation/support/RSocketMessageHandler.java]: `metadataMimeType` should actually be `metaMimeType`.

I don't have "permission to push to this repository", so I'll simply perform the work locally.

**Comment by sbrannen** [comment]
Resolved in 60a7092977eff8a967b6783ab3aa1f140dbe4a4f

**Pull request closed without merging**